tests/unit/test_kwargs/test_multipart_data.py

```diff
@@ -597,3 +618,100 @@ def test_invalid_multipart_raises_client_error() -> None:
             headers={"Content-Type": "multipart/form-data; charset=utf-8; boundary=20b303e711c4ab8c443184ac833ab00f"},
         )
         assert response.status_code == HTTP_400_BAD_REQUEST
+
+
+# Test for GitHub issue #4204: Empty strings in multipart/form-data should be preserved
+def test_empty_strings_preserved_in_multipart_forms() -> None:
+    """Test that empty strings are preserved in multipart forms and not converted to None.
+
+    This test addresses GitHub issue #4204 where empty strings in multipart/form-data
+    requests were being converted to None instead of being preserved as empty strings.
+    The test focuses on core multipart parsing behavior, not serialization library specifics.
```

Comment:
```suggestion
    This test addresses GitHub issue #4204 where empty strings in multipart/form-data
    requests were being converted to None instead of being preserved as empty strings.
    The test focuses on core multipart parsing behavior, not serialization library specifics.
```
this doesn't really add anything, a short comment and a link to the issue is fine